source/ast/Lookup.cpp

```diff
@@ -1827,12 +1830,27 @@ void Lookup::unqualifiedImpl(const Scope& scope, std::string_view name, LookupLo
                     locationGood = true;
                     break;
                 case SymbolKind::Parameter:
-                case SymbolKind::Specparam:
+                case SymbolKind::Specparam: {
```

Comment:
Probably the existing behavior for params can just be removed; I think I added that when other mechanisms for resolving recursive parameter definitions didn't exist. You can just merge all of the params + port / net / variable cases and handle them uniformly.